ci-operator/step-registry/ipi/conf/azure/default/ipi-conf-azure-default-commands.sh

```diff
@@ -0,0 +1,116 @@
+    echo "Getting inital release image from build farm imagestream: ${tmp_release_image_initial}"
+fi
+# For some ci upgrade job (stable N -> nightly N+1), RELEASE_IMAGE_INITIAL and 
+# RELEASE_IMAGE_LATEST are pointed to different imgaes, RELEASE_IMAGE_INITIAL has 
+# higher priority than RELEASE_IMAGE_LATEST
+TESTING_RELEASE_IMAGE=""
+if [[ -n ${tmp_release_image_initial:-} ]]; then
+    TESTING_RELEASE_IMAGE=${tmp_release_image_initial}
+else
+    TESTING_RELEASE_IMAGE=${RELEASE_IMAGE_LATEST_FROM_BUILD_FARM}
+fi
+echo "TESTING_RELEASE_IMAGE: ${TESTING_RELEASE_IMAGE}"
+
+dir=$(mktemp -d)
+pushd "${dir}"
+cp ${CLUSTER_PROFILE_DIR}/pull-secret pull-secret
+oc registry login --to pull-secret
+version=$(oc adm release info --registry-config pull-secret ${TESTING_RELEASE_IMAGE} --output=json | jq -r '.metadata.version' | cut -d. -f 1,2)
+major_version=$( echo "${version}" | awk --field-separator=. '{print $1}' )
+minor_version=$( echo "${version}" | awk --field-separator=. '{print $2}' )
+echo "get ocp version: ${version}"
+rm pull-secret
+popd
+
+CONFIG="${SHARED_DIR}/install-config.yaml"
+
+REGION="${LEASED_RESOURCE}"
+echo "Azure region: ${REGION}"
+
+workers=${COMPUTE_NODE_REPLICAS:-3}
+if [ "${COMPUTE_NODE_REPLICAS}" -le 0 ]; then
+  workers=0
+fi
+
+cat >> "${CONFIG}" << EOF
+baseDomain: ${BASE_DOMAIN}
+platform:
+  azure:
+    region: ${REGION}
+controlPlane:
+  architecture: ${OCP_ARCH}
+  name: master
+  platform: {}
+compute:
+- architecture: ${OCP_ARCH}
+  name: worker
+  replicas: ${workers}
+  platform: {}
+EOF
+
+if [ -z "${OUTBOUND_TYPE}" ]; then
+  echo "Outbound Type is not defined"
+else
+  if [ X"${OUTBOUND_TYPE}" == X"UserDefinedRouting" ] || [ X"${OUTBOUND_TYPE}" == X"NatGateway" ]; then
+    echo "Writing 'outboundType: ${OUTBOUND_TYPE}' to install-config"
+    PATCH="${SHARED_DIR}/install-config-outboundType.yaml.patch"
+    cat > "${PATCH}" << EOF
+platform:
+  azure:
+    outboundType: ${OUTBOUND_TYPE}
+EOF
+    yq-go m -x -i "${CONFIG}" "${PATCH}"
+  else
+    echo "${OUTBOUND_TYPE} is not supported yet" || exit 1
+  fi
+fi
+
+# User tags went GA in 4.14. In 4.14+, tag resources with an expiration date to facilitate cleanup.
+if (( minor_version > 13 && major_version == 4 )); then
+  expiration_date=$(date -d '8 hours' --iso=minutes --utc)
+  printf 'Setting user tag expirationDate: %s\n' "${expiration_date}"
+  yq-go write -i "${CONFIG}" "platform.azure.userTags.expiration_date" "${expiration_date}"
+fi
+
+REQUIRED_OCP_VERSION="4.12"
+isOldVersion=true
+if [ -n "${version}" ] && [ "$(printf '%s\n' "${REQUIRED_OCP_VERSION}" "${version}" | sort --version-sort | head -n1)" = "${REQUIRED_OCP_VERSION}" ]; then
+  isOldVersion=false
+fi
+
+PUBLISH=$(yq-go r "${CONFIG}" "publish")
+echo "publish: ${PUBLISH}"
+echo "is Old Version: ${isOldVersion}"
+if [ ${isOldVersion} = true ] || [ -z "${PUBLISH}" ] || [ X"${PUBLISH}" == X"External" ]; then
```

Comment:
I think we drop the if block here, because we want a pretty minimal install-config.yaml in this testing.

Reply:
Looks like the `baseDomainResourceGroupName` parameter is generated by default in the install-config, here's one example install-config created manually from the `openshift-install create install-config` command.
```
platform:
  azure:
    baseDomainResourceGroupName: os4-common
    cloudName: AzurePublicCloud
    outboundType: Loadbalancer
    region: eastus2
```

@jinyunma do you know if we can run a normal IPI installation without specifying the `baseDomainResourceGroupName`?  and maybe also `outboundType`, looks like `outboundType` would be "Loadbalancer" if we don't set it?

Reply:
Ah, I remember it now.
When < 4.12, whatever private or public cluster install, `baseDomainResourceGroupName` is always required, there is a known issue on private cluster install.
When >=4.12, when it is a private cluster, it can be skipped, when it is a external cluster, we have to set it.

So in this case, I think we can directly set `baseDomainResourceGroupName` in install-config.yaml with skipping the version check.

Reply:
`baseDomainResourceGroupName` is required for common IPI installation, and not required when installing private cluster. For default configuration test, as we only test common IPI installation, I think we can always specify `baseDomainResourceGroupName` here.

For `outboundType`, I think it can be skipped in install-config, because the default value is `Loadbalancer`, I just checked that there is no error reported when creating manifests, and outboundType is set to `Loadbalancer` in "cluster-config.yaml". Let me do more test when creating cluster without specifying `outboundType`

Reply:
> Let me do more test when creating cluster without specifying outboundType

Installation is running into stage of "waiting for bootstrapping to complete", external load balancer is created successful, so `outboundType` can also be skipped here.

Reply:
OK, dropped both of `baseDomainResourceGroupName` and `outboundType` condition checks, set `baseDomainResourceGroupName` in the install-config directly and omit the `outboundType`